### ISSUE TITLE
Cargo clippy crashes in root of workspace

### ISSUE THREAD

**Issue by sanmai-NL** [rust-lang-nursery/rust-clippy#2105]
```sh
env RUST_BACKTRACE=1 cargo +nightly clippy
```
```
thread 'main' panicked at 'could not find parent of path /', /home/sanmai/.cargo/bin/registry/src/github.com-1ecc6299db9ec823/clippy-0.0.165/src/main.rs:250:51
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:397
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:577
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:538
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:522
   7: cargo_clippy::main
   8: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:99
   9: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:459
             at /checkout/src/libstd/panic.rs:361
             at /checkout/src/libstd/rt.rs:59
  10: __libc_start_main
  11: _start
```

## Meta

```sh
cargo +nightly clippy -V
```
```
0.0.165
```
See also #1515

**Comment by oli-obk** [comment]
I cannot reproduce. On which crate are you running it and from which directory?

**Comment by sanmai-NL** [comment]
The output of `pwd` is `/home/sanmai/devel/gitlab.com/Structure-Systems/SE/Rust`. I am running it from a workspace, and it contains unpublished crates.

**Comment by oli-obk** [comment]
So I'm assuming the `Cargo.toml` is in that folder?

How are the other crates positioned relative to it? We have a workspace test for


* workspace_root
    * subcrate

And that works from both directories + any subdirectory of either.

**Comment by radix** [comment]
I'm getting this too, in github.com/radix/pandt. I'm using windows 10 and running it from the root of the repository.

```
C:\Users\radix\Projects\pandt [master ↑]> cargo clippy
thread 'main' panicked at 'could not find parent of path \\?\C:\', C:\Users\radix\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy-0.0.165\src/main.rs:250:51
```

**Comment by sanmai-NL** [comment]
@oli-obk: I have `.cargo` directories at multiple layers, one in the workspace root dir and one in each workspace member dir, (and one globally in my home dir). Does this help track down the problem?

**Comment by lukaramu** [comment]
Just ran into this crash myself. My (complete!) workspace Cargo.toml looked something like this:
```
[workspace]

members = [
    "foo",
    "bar",
    "baz",
]
```

Since @oli-obk said that there *was* a workspace test, I looked at that and found that the test workspace had a `[package]` section while mine did not. After prepending that to my workspace `Cargo.toml`, it doesn't crash anymore, instead saying `error: Could not obtain cargo metadata.` (presumably because I still have neither `[lib]` nor `[[bin]]` sections in my workspace Cargo.toml?? This is pure speculation).

While this makes the crash go away, it doesn't really make sense for me to add a `[package]` section to my workspace's Cargo.toml since it really isn't a package -- it would be nice if I could run `cargo +nightly clippy` in my workspace root to just check all its members with `clippy`.

**Comment by samueltardieu** [comment]
You can also reproduce it with:

``` bash
% git clone https://github.com/hashmismatch/packed_struct.rs
% cd packed_struct.rs
% cargo clippy
thread 'main' panicked at 'could not find parent of path /', src/main.rs:124:48
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by phansch** [comment]
Looks like this is fixed with 0.0.191 (possibly in https://github.com/rust-lang-nursery/rust-clippy/pull/2585). I'm going ahead and close this because the offending code has been removed and is now handled by `cargo check`.

@sanmai-NL Feel free to re-open if you still get this crash with 0.0.191.
